### PR TITLE
docs: add dev setup guide and make package scripts cross-platform (Windows)

### PR DESCRIPTION
**Pull request by buildwithhamzat**
## What does this PR do?

Adds a repo-native **QA execution layer** plus a couple of dev-tooling fixes.

**QA system** (test cases live in the repo; bugs live in GitHub Issues - there is no
duplicate tracker to keep in sync):
- `FEATURE_AUDIT.md` - the feature **map**: every feature, its user flow, and edge cases
  with `file:line` citations.
- `qa/README.md` - the **workflow**: severity/difficulty rubrics, label setup, and a static
  feature -> flow coverage map proving all 42 audited features have a test plan.
- `qa/flows/01..13` - the **test plan**: 13 per-flow happy-path checklists (signup,
  course authoring, people/invites, student -> certificate, AI assistant, programs,
  community, org public site, media/import, org settings, public API/MCP, SSO/licensing,
  billing). The checkboxes are the execution record.
- `.github/ISSUE_TEMPLATE/qa-bug.yml` - a **QA Bug** issue template requiring severity,
  difficulty, area, repro steps, and a screen recording.

**Dev tooling:**
- `db:studio` and `docker:build` scripts (`package.json`, `packages/db/package.json`).
- Fix `pnpm db:reset` on Windows: spawn the `pnpm.cmd` shim with `shell: true`
  (Node's CVE-2024-27980 fix throws `EINVAL` when spawning `.cmd`/`.bat` without a shell).

No issue to close - this is internal QA infrastructure.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue) - Windows `db:reset`
- [x] Chore (refactoring code, technical debt, workflow improvements) - QA docs + tooling
- [ ] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change adds a new database migration
- [x] This change requires a documentation update

## How should this be tested?

- Render `qa/README.md` and the 13 `qa/flows/*.md` files on GitHub: links resolve, the
  feature -> flow coverage map is present, and no reference to a removed tracker remains.
- In GitHub's **New issue** UI, pick **QA Bug**: confirm severity/difficulty/area/repro and
  the screen-recording field are required.
- Run `pnpm db:studio` and `pnpm docker:build` from the repo root - both resolve and run.
- On **Windows**, run `pnpm db:reset` - it completes instead of throwing `EINVAL`.

### Required

- [x] Filled out the "How to test" section in this PR
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand bits
- [ ] Ran `pnpm build`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [ ] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [x] Updated the ClassroomIO Docs if changes were necessary

<h3>Greptile Summary</h3>

This PR adds a repo-native QA layer (13 per-flow test checklists, a QA Bug issue template, and a feature-to-flow coverage map) and fixes several cross-platform dev-tooling issues: Windows `pnpm db:reset` EINVAL, the Unix-only postinstall script, `rm -rf`/`cp` usages replaced with `rimraf`/Node.js builtins, and localhost-bound Docker port bindings.

- **Windows fixes**: `shell: process.platform === 'win32'` added to `spawn` in both `db-reset.ts` and `db-setup.ts`; `postinstall` rewritten as an ESM `.mjs` script replacing the Unix-only `test -d .git && lefthook install || true` one-liner.
- **Cross-platform build scripts**: `rm -rf` → `rimraf` across all workspace packages, `cp` → `node --input-type=commonjs -e require('fs').copyFileSync(...)` in `apps/embeds` (properly overrides `"type": "module"` via `--input-type`).
- **Docker**: postgres and redis ports bound to `127.0.0.1` so host tools (`pnpm db:setup`, `pnpm api:dev`) can reach them without LAN exposure; `db-init` service removed, schema setup now runs inside the `api` container.

<h3>Confidence Score: 4/5</h3>

Safe to merge for most environments; the lockfile regeneration that strips `libc` fields from native Linux binary entries remains in the diff and could cause Alpine/musl installs to pick the wrong binary variant at runtime.

The core code changes — Windows `shell` flag, ESM-safe file copy, cross-platform postinstall, localhost Docker ports — are all correct and well-scoped. The unresolved concern is the pnpm-lock.yaml diff that removes `libc: [glibc]`/`libc: [musl]` discriminators from dozens of optional native packages (`@img/sharp-*`, `@rollup/rollup-linux-*`, `@tailwindcss/oxide-linux-*`, etc.); on musl-based Docker images this can cause the wrong native binary variant to be installed, leading to `Exec format error` at runtime. That lockfile change appears to be a side-effect of running `pnpm install` with a different pnpm version rather than an intentional edit.

pnpm-lock.yaml — the removal of `libc` fields from native binary package entries warrants a second look to confirm it was regenerated with the correct pnpm version.

<h3>Important Files Changed</h3>

| Filename | Overview |
|----------|----------|
| apps/embeds/package.json | Replaces `cp` with `node --input-type=commonjs -e` (ESM-safe copy) and `rm -rf dist` with `rimraf dist`; both the `--input-type=commonjs` flag and the implicit `rimraf` root-binary dependency are new. |
| packages/db/src/scripts/db-reset.ts | Adds `shell: process.platform === 'win32'` to the `spawn` call so `pnpm.cmd` can be executed on Windows without hitting the CVE-2024-27980 EINVAL restriction. |
| packages/db/src/scripts/db-setup.ts | Identical Windows spawn fix as db-reset.ts — `shell: process.platform === 'win32'` added to the pnpm.cmd spawn call. |
| scripts/postinstall.mjs | New cross-platform postinstall: replaces the Unix-only `test -d .git && lefthook install || true` one-liner with an ESM script using `existsSync` + `execSync` wrapped in try/catch. |
| docker/docker-compose.yaml | Adds `127.0.0.1:5432:5432` and `127.0.0.1:6379:6379` port bindings to postgres and redis, making them reachable from host tools while keeping them off the LAN. |
| package.json | Adds `rimraf` as a root devDependency, adds `db:studio` proxy script, switches `clean` from `find/rm` to `rimraf --glob`, and replaces the inline postinstall shell one-liner with `scripts/postinstall.mjs`. |
| pnpm-lock.yaml | Adds rimraf@6.1.3 and its dependencies; also removes `libc` fields from many native Linux binary package entries (pre-existing concern already flagged in review). |
| packages/db/README.md | Updates demo-account table: corrects admin email from `test@test.com` to `admin@test.com` (matching users.json) and updates password label to `123456`; adds `.env.example` bootstrap instructions. |
| packages/db/.env.example | New file providing a template DATABASE_URL pointing to the local docker-compose postgres instance. |
| apps/docs/content/docs/quickstart/self-hosting.mdx | Updates seed command from removed `db-init` service to the `api` container; replaces literal placeholder secrets with `<generate-with-openssl-rand-hex-32>` instructions. |
| docker/docs/SELF_HOST.md | Removes `db-init` from stack description and command reference; updates seed command to use `api` container. |

</details>

<sub>Reviews (6): Last reviewed commit: ["fix: update postinstall script"](https://github.com/classroomio/classroomio/commit/811e5b94345e848adca5fc4c7175d86b30299f2d) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=36202114)</sub>

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/qa-bug.yml [hunk 1]
```diff
@@ -0,0 +1,132 @@
+name: 'QA Bug'
```
Comment (rotimi-best) [inline .github/ISSUE_TEMPLATE/qa-bug.yml]: Replace the current `classroomio/classroomio/.github/ISSUE_TEMPLATE/bug-report.yml` with this.

File: .github/ISSUE_TEMPLATE/qa-bug.yml [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+description: Internal QA bug found while walking a flow checklist. Include a screen recording.
```
Comment (rotimi-best) [inline .github/ISSUE_TEMPLATE/qa-bug.yml]: Let's remove the word 'Internal' from this.

File: .github/ISSUE_TEMPLATE/qa-bug.yml [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        - Medium (hours, single area)
+        - Hard (cross-cutting / unknown root cause)
+    validations:
+      required: true
+  - type: dropdown
+    id: area
+    attributes:
+      label: Area
+      options:
+        - auth
```
Comment (rotimi-best) [inline .github/ISSUE_TEMPLATE/qa-bug.yml]: Add an area for backend related bugs

File: .github/ISSUE_TEMPLATE/qa-bug.yml [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+        - 'Self-hosted only'
+        - 'Not yet verified in the other mode'
+  - type: checkboxes
+    attributes:
+      label: Before filing
+      options:
+        - label: I searched existing issues and this is not a duplicate.
+        - label: I attached a screen recording.
+        - label: I will apply the matching `severity:*` and `difficulty:*` labels.
+        - label: I linked this issue # in the relevant qa/flows step.
```
Comment (rotimi-best) [inline .github/ISSUE_TEMPLATE/qa-bug.yml]: I really love this, thanks

File: apps/embeds/package.json [hunk 5]
```diff
@@ -7,8 +7,8 @@
     "dev": "vite dev --port 5180",
     "prebuild": "pnpm --filter @cio/ui run build",
     "build:question-type-picker": "vite build --mode question-type-picker",
-    "build:course-widget": "vite build --mode course-widget && cp src/widgets/course-widget/embed.html dist/course-widget/embed.html",
-    "build": "rm -rf dist && pnpm build:question-type-picker && pnpm build:course-widget",
+    "build:course-widget": "vite build --mode course-widget && node -e \"require('fs').copyFileSync('src/widgets/course-widget/embed.html','dist/course-widget/embed.html')\"",
```
[the remark this answers is not in the document]
Comment (rotimi-best) [reply apps/embeds/package.json]: You should apply this, it is a good suggestion.

File: scripts/format-check.mjs [hunk 6]
```diff
@@ -0,0 +1,54 @@
+// Cross-platform Prettier check for the pre-commit hook (lefthook.yml).
```
Comment (rotimi-best) [inline scripts/format-check.mjs]: This is updated in main, pull main and try it out for any windows related issues.

File: package.json [hunk 7]
```diff
@@ -6,13 +6,14 @@
     "start": "turbo run start",
     "dev": "turbo prepare --concurrency 13 && turbo run dev",
     "lint": "turbo run lint",
-    "clean": "turbo clean && find . -name 'node_modules' -type d -prune -exec rm -rf '{}' +",
+    "clean": "turbo clean && rimraf --glob \"**/node_modules\"",
     "clean:pnpm": "pnpm clean && pnpm store prune",
     "format": "turbo run format",
-    "format:check": "git diff --name-only -z --diff-filter=d HEAD~1 | xargs -0 -I{} sh -c '[ ! -L \"{}\" ] && printf \"%s\\0\" \"{}\"' | xargs -0 prettier --check --ignore-unknown",
-    "postinstall": "test -d .git && lefthook install || true",
+    "format:check": "node scripts/format-check.mjs",
```
Comment (rotimi-best) [inline package.json]: This is also changed in main. please pull from main

File: package.json [hunk 8]
```diff
@@ -23,7 +24,9 @@
     "embeds:dev": "pnpm --filter=@cio/embeds --filter=@cio/ui run dev",
     "storybook:dev": "pnpm --filter=@cio/storybook --filter=@cio/ui --filter=@cio/question-types run dev",
     "mcp:build": "pnpm --filter=@classroomio/mcp run build",
-    "db:push": "pnpm --filter @cio/db db push"
+    "db:push": "pnpm --filter @cio/db db push",
+    "docker:start": "docker compose -f docker/docker-compose.yaml up -d postgres redis",
+    "docker:build": "docker compose -f docker/docker-compose.yaml up --build -d postgres redis"
```
Comment (rotimi-best) [inline package.json]: You should keep docker related changes in #666

**Review by rotimi-best — CHANGES_REQUESTED** [review]
This is great, left a few comments.

Migrate the qa docs related changes from this so we can merge the code changes in

**Review by rotimi-best — APPROVED** [review]
Great work, LGTM
